app/assets/javascripts/channels/synchronized_editor_channel.js

```diff
@@ -24,12 +24,12 @@ $(document).on('turbolinks:load', function () {
         received(data) {
           // Called when there's incoming data on the websocket for this channel
           if (current_user_id !== data['current_user_id']) {
```

Comment:
This condition prevents a single user from opening the same exercise in two windows. Further, we should not just use the user ID, since the user type needs to match as well.

Potentially, we could also switch to a random identifier (such as a UUID for each page load), and use that one as identifier. It should resolve those issues, I assume.